common/data_refinery_common/models/models.py

```diff
@@ -916,7 +918,9 @@ def has_blocking_jobs(self, own_processor_id=None) -> bool:
         # If the file has a processor job that should not have been
         # retried, then it still shouldn't be retried.
         # Exclude the ones that were aborted.
-        no_retry_processor_jobs = self.processor_jobs.filter(no_retry=True).exclude(abort=True)
+        no_retry_processor_jobs = self.processor_jobs.filter(
+            no_retry=True, worker_version=SYSTEM_VERSION
+        ).exclude(abort=True)
```

Comment:
I'm wondering if we should also remove `exclude(abort=True)` which was an attempt to achieve the same?

Reply:
If the version stays the same ie the job is retried without a re-deploy then excluding `abort=True` would return 0 which to would mean the job is retried. This code only says to respect `no_retry` for the current version.